client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java

```diff
@@ -268,8 +268,14 @@ private static Request getStyleRequest(String method, GetRequest getRequest) {
     }
 
     static Request sourceExists(GetRequest getRequest) {
-        Request request = new Request(HttpHead.METHOD_NAME, endpoint(getRequest.index(), getRequest.type(), getRequest.id(), "_source"));
-
+        String optionalType = getRequest.type();
+        String endpoint;
+        if (optionalType == null || optionalType.equals(MapperService.SINGLE_MAPPING_NAME)) {
```

Comment:
Two very small comments: the indentation is a bit off, and I don't think we need to check if the type is null because `GetRequest#validate` ensures that it isn't.